arches/app/models/resource.py

```diff
@@ -71,11 +71,13 @@ def __init__(self, *args, **kwargs):
         self.tiles = []
 
     def get_descriptor(self, descriptor):
-        module = importlib.import_module("arches.app.functions.primary_descriptors")
-        PrimaryDescriptorsFunction = getattr(module, "PrimaryDescriptorsFunction")()
-        functionConfig = models.FunctionXGraph.objects.filter(graph_id=self.graph_id, function__functiontype="primarydescriptors")
-        if len(functionConfig) == 1:
-            return PrimaryDescriptorsFunction.get_primary_descriptor_from_nodes(self, functionConfig[0].config[descriptor])
+        function_config = models.FunctionXGraph.objects.filter(graph_id=self.graph_id, function__functiontype="primarydescriptors")
+        if len(function_config) == 1:
+            module = importlib.import_module(function_config[0].config["module"])
```

Comment:
This could have originally been:
```
            func = function_config[0].function.get_class_module()()
            return func.get_primary_descriptor_from_nodes(
                 self, function_config[0].config["descriptor_types"][descriptor]
            )
```
I'm not sure if this was previously hard-coded for performance issues. the config.module & config.classname JSON attributes could potentially be collapsed into packagename if the above is too heavy. Could be something like this:
```
            module = importlib.import_module(function_config[0].config["package"]+"."+function_config[0].modulename)
            primary_descriptors_function = getattr(module, function_config[0].classname)()
            return primary_descriptors_function.get_primary_descriptor_from_nodes(
                self, function_config[0].config["descriptor_types"][descriptor]
            )
```
Same holds true for api.py

Reply:
I took another look at this and realized that the original code is using the function object's get_class_module and immediately creating and instance of the module. In this way the module location and class name don't come from the config, but rather the classname and modulename columns of the Function model. This is also the pattern that is used with Functions that are not primary descriptors. 

I think, however, the config would still have to come from the FunctionXGraphs instance. So something like this:

```
def get_descriptor(self, descriptor):
    function = models.Function.objects.filter(graphmodel=self.graph_id, functiontype="primarydescriptors")
    if len(function) == 1:
        module = function[0].get_class_module()()
        config = models.FunctionXGraph.objects.get(graph_id=self.graph_id, function=function[0].functionid).config
        return module.get_primary_descriptor_from_nodes(self, config["descriptor_types"][descriptor])
    else:
        return "undefined"
```

Reply:
@chiatt - How about this to take advantage of the single database call and still using the get_class_module() on the related function?
```
    def get_descriptor(self, descriptor):
        graph_function = models.FunctionXGraph.objects.filter(
            graph_id=self.graph_id, function__functiontype="primarydescriptors"
        ).select_related("function")
        if len(graph_function) == 1:
            module = graph_function[0].function.get_class_module()()
            return module.get_primary_descriptor_from_nodes(
                self, graph_function[0].config["descriptor_types"][descriptor]
            )
        else:
            return "undefined"
```
I totally agree about getting rid of the module and class name in the config, it was bugging me too. This gets hit a lot when performing data imports so it might be worth keeping the number of DB calls down.